frontend/src/components/Combobox/Combobox.tsx

```diff
@@ -0,0 +1,278 @@
+import { useCallback, useMemo, useState } from 'react'
+import { usePopper } from 'react-popper'
+import {
+  Box,
+  Flex,
+  FormControlOptions,
+  forwardRef,
+  InputGroup,
+  List,
+  ListItem,
+  StylesProvider,
+  useFormControlProps,
+  useMultiStyleConfig,
+} from '@chakra-ui/react'
+import { useCombobox } from 'downshift'
+
+import Input from '~components/Input'
+
+import { ComboboxClearButton } from './ComboboxClearButton'
+import { DropdownItem } from './DropdownItem'
+import { LabelIcon } from './LabelIcon'
+import { ToggleChevron } from './ToggleChevron'
+import { ComboboxItem } from './types'
+import {
+  defaultFilter,
+  itemToIcon,
+  itemToLabelString,
+  itemToValue,
+} from './utils'
+
+export interface ComboboxProps<Item = ComboboxItem, Value = string>
```

Comment:
aren't we being inconsistent about `Value` being a generic type? for instance, despite the fact that `Value` is generic, `itemToValue` always returns a `string`. imo there's no need to overengineer this, we can just let it be `string`